pkg/encoding/sanify.go

```diff
@@ -0,0 +1,49 @@
+package encoding
+
+import (
+	"github.com/pkg/errors"
+	"github.com/surajssd/kapp/pkg/spec"
+)
+
+func fixApp(app *spec.App) error {
+
+	// fix app.Services
+	for i, service := range app.Services {
+		if err := fixService(&service, app); err != nil {
+			return errors.Wrapf(err, "Unable to fix service: %v", service.Name)
+		}
+		app.Services[i] = service
+	}
+
+	// fix app.PersistentVolumes
+	for i, pVolume := range app.PersistentVolumes {
+		if err := fixPersistentVolume(&pVolume, app); err != nil {
+			return errors.Wrapf(err, "Unable to fix persistentVolume: %v", pVolume.Name)
+		}
+		app.PersistentVolumes[i] = pVolume
+	}
+
+	return nil
+}
+
+func fixService(service *spec.ServiceSpecMod, app *spec.App) error {
```

Comment:
looping is cheap ;-)
Someone could easily make a mistake and pass two different App object into this function.
We don't have to worry about performance here. Focusing on code readability and simplicity is more important.

But I don't want to argue about stupid stuff, so we can keep it as it is if you want.